src/Strategy/Gradle.hs

```diff
@@ -72,35 +101,47 @@ discover ::
   ) =>
   Path Abs Dir ->
   m [DiscoveredProject run]
-discover dir = context "Gradle" $ do
-  projects <- context "Finding projects" $ findProjects dir
-  pure (map mkProject projects)
-
-pathToText :: Path ar fd -> Text
-pathToText = T.pack . toFilePath
-
+discover dir =
+  context "Gradle" $
+    (fmap . fmap) mkProject $ context "Finding projects" $ findProjects dir
```

Comment:
I know the do-notation version isn't pleasant, but this feels less readable to me